ext/AWSCRTExt.jl

```diff
@@ -3,41 +3,31 @@ module AWSCRTExt
 import AWSCRT, MQTT
 
 struct MQTT.MQTTConnection <: MQTT.AbstractConnection
-    client::AWSCRT.MQTTClient
     connection::AWSCRT.MQTTConnection
+    endpoint::String
     port::Int
     id::String
-    will::AWSCRT.Will
-
-    MQTT.MQTTConnection(client, port, id, will) = new(client, AWSCRT.MQTTConnection(client), port, id, will)
+    connect_kwargs::Dict{Symbol,Any}
 end
 
+MQTT.MQTTConnection(connection, endpoint, port, id; connect_kwargs = Dict()) =
```

Comment:
The idea here is to support all the possible connect kwargs without blowing up the code in here, since there are many and it will keep growing https://octogonapus.github.io/AWSCRT.jl/dev/#AWSCRT.connect-Tuple{MQTTConnection,%20String,%20Integer,%20String}